pkg/plugin/framework_methods.go

```diff
@@ -0,0 +1,610 @@
+	// * proposedNotInLocalState are the pods in tmpNode but not oldNode
+	//
+	// We'll use (another) Speculatively() to simultaneously show all these, plus the state
+	// resulting from adding the Pod to filter.
+	var canAddToNode bool
+	tmpNode.Speculatively(func(n *state.Node) (commit bool) {
+		n.AddPod(filterPod)
+		canAddToNode = !n.OverBudget()
+
+		var msg string
+		if canAddToNode {
+			msg = "Allowing Pod placement onto this Node"
+		} else {
+			msg = "Rejecting Pod placement onto this Node"
+		}
+		logger.Info(
+			msg,
+			zap.Object("Node", oldNode),
+			zap.Object("FilterNode", tmpNode),
+			zap.Object("FilterNodeWithPod", n),
+			zap.Object("Pod", filterPod),
+			zap.Any("LocalPodsNotInFilterState", localNotInProposed),
+			zap.Any("FilterPodsNotInLocalState", proposedNotInLocalState),
+		)
+
+		return false // don't commit. Doesn't really matter because we're operating on the temp node.
+	})
+	return canAddToNode
+}
+
+// Score allows our plugin to express which nodes should be preferred for scheduling new pods onto
+//
+// Even though this function is given (pod, node) pairs, our scoring is only really dependent on
+// values of the node. However, we have special handling for when the pod no longer fits in the node
+// (even though it might have during the Filter plugin) - we can't return a failure, because that
+// would cause *all* scheduling of the pod to fail, so we instead return the minimum score.
+//
+// The scores might not be consistent with each other, due to ongoing changes in the node. That's
+// ok, because nothing relies on strict correctness here, and they should be approximately correct
+// anyways.
+//
+// Score implements framework.ScorePlugin.
+func (e *AutoscaleEnforcer) Score(
+	ctx context.Context,
+	_state *framework.CycleState,
+	pod *corev1.Pod,
+	nodeName string,
+) (_ int64, status *framework.Status) {
+	ignored := e.state.config.ignoredNamespace(pod.Namespace)
+
+	e.metrics.IncMethodCall("NormalizeScore", pod, ignored)
+	defer func() {
+		e.metrics.IncFailIfnotSuccess("NormalizeScore", pod, ignored, status)
+	}()
+
+	logger := e.logger.With(
+		zap.String("method", "NormalizeScore"),
+		reconcile.ObjectMetaLogField("Pod", pod),
+	)
+
+	logger.Info("Handling Score request", logFieldForNodeName(nodeName))
+
+	if status := e.checkSchedulerName(logger, pod); status != nil {
+		return framework.MinNodeScore, status
+	}
+
+	podState, err := state.PodStateFromK8sObj(pod)
+	if err != nil {
+		msg := "Error extracting local information for Pod"
+		logger.Error(msg, zap.Error(err))
+		return framework.MinNodeScore, framework.NewStatus(
+			framework.UnschedulableAndUnresolvable,
+			fmt.Sprintf("%s: %s", msg, err.Error()),
+		)
+	}
+
+	e.state.mu.Lock()
+	defer e.state.mu.Unlock()
+
+	ns, ok := e.state.nodes[nodeName]
+	if !ok {
+		msg := "Node not found in local state"
+		logger.Error(msg)
+		status := framework.NewStatus(framework.Error, fmt.Sprintf("%s: %s", msg, err.Error()))
```

Comment:
issue (blocking): `err` is incorrectly used and is always `nil` here

suggestion:
```suggestion
		status := framework.NewStatus(framework.Error, msg)
```

Reply:
Fixed in 3a9f02cb6ce1aecb1193f9ba31cb00ad8aa085af